index.js

```diff
@@ -50,7 +50,10 @@ http.Agent = function () {};
 http.Agent.defaultMaxSockets = 4;
 
 var xhrHttp = (function () {
-    if (typeof window === 'undefined') {
+    if (typeof self !== 'undefined' && self.XMLHttpRequest) {
```

Comment:
How about this:
```js
if (typeof XMLHttpRequest !== 'undefined') {
    return XMLHttpRequest;
}
```
And then remove the second check for `window.XMLHttpRequest` below.